src/main/java/software/amazon/qldb/QldbSession.java

```diff
@@ -128,6 +128,8 @@ boolean isClosed() {
                 if (executionContext.retryAttempts() >= retryPolicy.maxRetries()) {
                     throw sce;
                 }
+            } catch (Exception e) {
+                noThrowAbort(transaction);
```

Comment:
You need to re-throw the exception here after abort.

Reply:
Good catch - fixed in update